### PR TITLE
fix: offline deleted heading styles and link padding

### PR DESCRIPTION
**Pull request by Nikhil-Vats**
### Details
This PR fixes the following issues -
1. Deleted headings in offline mode looked like underlined instead of strikethrough.
2. When only a link is added in the header text, the app adds extra padding above the link which has the cursor as a pointer.

The first one was a regression from #16526, this PR fixes that issue using another way.

### Fixed Issues

$ https://github.com/Expensify/App/issues/18658
$ https://github.com/Expensify/App/issues/17488
$ https://github.com/Expensify/App/issues/18658#issuecomment-1560205815

### Tests

For testing #17488 (padding above header links) -
1. Open the app in Chrome/safari on your desktop.
2. Open a chat.
3. Send any link as header text. eg: # google.com
4. Verify that there is no extra padding above links and that links are only clickable (with the cursor as a pointer) when the cursor is over the text not above it.

You can also follow the same steps on the desktop app on MacOS to verify that it is working successfully there as well.

For testing #16526 (cursor shown in empty space for links that span multiple lines) -
1. Post a long hyperlink as a comment.
e.g link : [Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer ornare dignissim nunc, eleifend molestie dui tempus non. Proin semper eu metus sit amet feugiat. Sed turpis augue, pellentesque sed accumsan ornare, malesuada in ligula. Aliquam porta condimentum varius. Morbi id lorem felis.](https://google.com/)
2. Hover over the hyperlink after posting the comment.
3. On web and desktop platforms pointer cursor should be shown only when the mouse cursor is over the hyperlink and not when it is over the empty space near the line endings.
4. Regular cursor should be shown while hovering over the empty area next to the hyperlink.
5. On other platforms, the comment should be shown exactly as it was rendered earlier.
6. All other comment types should render exactly the same as before.
7. Same behaviour is applicable for links posted as a reply to a comment via email.

- [x] Verify that no errors appear in the JS console

### Offline tests
Issue #18658 must be tested in offline mode. Follow the steps below -
1. Open the app in Chrome on your desktop.
2. Go to a chat and add a heading comment. eg: # Example
3. Go offline by switching off the connection.
4. Delete the message.
5. Verify that the message is shown with a strikethrough style (line over the message) and not underlined.

Note -
1. Repeat the steps above for the following messages (feel free to add more) -
Messages are separated by a new empty line. 
```
# text header

# header with link.com

# header with emoji 😇

# 😇😇

> # header inside quote with [link](google.com)

`# try header in code`

\```
multiline code (remove \ before three backticks above and below)
😅
link.com
\```

😋😆😇😇

[normal link](google.com)
```
2. Also repeat the steps for different attachments - images, PDFs, files without preview like CSVs, docs. Upload them in online mode, go offline and delete them. 
  a. After deleting attachments like images, PDFs, GIFs which have preview, their opacity should decrease and user should not be able to open them.
  b. For attachments without previews like CSV, docs, etc. A strikethrough should be shown similar to text comments. 
3. Add any attachments in offline mode. App should keep showing "Uploading attachment...". Delete the attachment. App should show a strikethrough over the text.

You can test this behaviour on a mobile browser, desktop browser, MacOS desktop app, iOS app, android app, etc using the same steps.

### QA Steps

<details>
<summary>
Follow the same steps listed in the section above. Expand this to see the same steps.
</summary>

For testing #17488 (padding above header links) -
1. Open the app in Chrome/safari on your desktop.
2. Open a chat.
3. Send any link as header text. eg: # google.com
4. Verify that there is no extra padding above links and that links are only clickable (with the cursor as a pointer) when the cursor is over the text not above it.

You can also follow the same steps on the desktop app on MacOS to verify that it is working successfully there as well.

For testing #16526 (cursor shown in empty space for links that span multiple lines) -
1. Post a long hyperlink as a comment.
e.g link : [Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer ornare dignissim nunc, eleifend molestie dui tempus non. Proin semper eu metus sit amet feugiat. Sed turpis augue, pellentesque sed accumsan ornare, malesuada in ligula. Aliquam porta condimentum varius. Morbi id lorem felis.](https://google.com/)
2. Hover over the hyperlink after posting the comment.
3. On web and desktop platforms pointer cursor should be shown only when the mouse cursor is over the hyperlink and not when it is over the empty space near the line endings.
4. Regular cursor should be shown while hovering over the empty area next to the hyperlink.
5. On other platforms, the comment should be shown exactly as it was rendered earlier.
6. All other comment types should render exactly the same as before.
7. Same behaviour is applicable for links posted as a reply to a comment via email.

### Offline tests
Issue #18658 must be tested in offline mode. Follow the steps below -
1. Open the app in Chrome on your desktop.
2. Go to a chat and add a heading comment. eg: # Example
3. Go offline by switching off the connection.
4. Delete the message.
5. Verify that the message is shown with a strikethrough style (line over the message) and not underlined.

Note -
1. Repeat the steps above for the following messages (feel free to add more) -
Messages are separated by a new empty line. 
```
# text header

# header with link.com

# header with emoji 😇

# 😇😇

> # header inside quote with [link](google.com)

`# try header in code`

\```
multiline code (remove \ before three backticks above and below)
😅
link.com
\```

😋😆😇😇

[normal link](google.com)
```
2. Also repeat the steps for different attachments - images, PDFs, files without preview like CSVs, docs. Upload them in online mode, go offline and delete them. 
  a. After deleting attachments like images, PDFs, GIFs which have preview, their opacity should decrease and user should not be able to open them.
  b. For attachments without previews like CSV, docs, etc. A strikethrough should be shown similar to text comments. 
3. Add any attachments in offline mode. App should keep showing "Uploading attachment...". Delete the attachment. App should show a strikethrough over the text.

You can test this behaviour on a mobile browser, desktop browser, MacOS desktop app, iOS app, android app, etc using the same steps.

</details>

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot for #18658 
image

Video for #16526 and #17488 

https://github.com/Expensify/App/assets/31413407/157e5cba-d0ab-490a-b456-2ee58029af2f

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/31413407/08a5301a-9211-4d70-a736-39fb901ad25e

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot for #18658
image

Video for #16526

https://github.com/Expensify/App/assets/31413407/5522105d-e395-4ce9-9e43-019f51d48d1a

</details>

<details>
<summary>Desktop</summary>

Screenshot for #18658 
Screenshot 2023-05-24 at 10 48 58 PM

Video for #16526 and #17488 

https://github.com/Expensify/App/assets/31413407/635337ef-77b3-4c73-aa63-e12c03dfc8bd

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/31413407/ced299ae-d7eb-4ac9-940c-cc443c3dca54

</details>

<details>
<summary>Android</summary>

Screenshot for #18658 
Screenshot 2023-05-25 at 10 52 46 AM

Video for #16526 and #17488 

https://github.com/Expensify/App/assets/31413407/705e3ce1-a795-4147-bbfe-2bafdfb77a4c

</details>

### CODE REVIEW

**Comment by Nikhil-Vats** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
Comment (Nikhil-Vats) [inline src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js:51]: @parasharrajat I plan to add a comment here which says that the `inline` prop is only used for the native environment.

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (Nikhil-Vats) [inline src/components/PressableWithSecondaryInteraction/index.js:87]: added this at the end, since `defaultPressableProps` were overwriting the `style` prop when it was in the first line as `defaultPressableProps` have `this.props.style` as one of the keys. Here, we add display inline to default props, if `inline` is true.

File: src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
Comment (Nikhil-Vats) [inline src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js:51]: not setting `display` to `inline` here as it will pass the style to the native component as well which doesn't support the `inline` value.

Comment (Nikhil-Vats) [reply src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js:51]: I tested all three issues and they were fixed on the respective platforms. I will go through the issue discussion on #16526 and check if there are any bugs.

File: src/components/PressableWithSecondaryInteraction/index.js
Comment (mollfpr) [reply src/components/PressableWithSecondaryInteraction/index.js:87]: @Nikhil-Vats This led to regression for menu items on the Web and Desktop.

<details>
<summary>web</summary>
Screenshot 2023-05-25 at 21 34 44



</details>

<details>
<summary>desktop</summary>

Screenshot 2023-05-25 at 21 36 38

</details>

Comment (mollfpr) [reply src/components/PressableWithSecondaryInteraction/index.js:87]: Another regression on the copy url clipboard popover.

<details>
<summary>DEV</summary>

Screenshot 2023-05-25 at 21 45 31

</details>

<details>
<summary>STAGING</summary>

Screenshot 2023-05-25 at 21 45 40

</details>

Comment (Nikhil-Vats) [reply src/components/PressableWithSecondaryInteraction/index.js]: @mollfpr both fixed. I was not handling style props which were functions.

File: src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
Comment (Nikhil-Vats) [reply src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js:51]: @parasharrajat I have read through your concerns on #16526 and tested out all kinds of message types - Attachments (with and without previews), and various markdown types like header, italic, bold, quote, code, code blocks, links, email comments etc. and it works as expected. 

`BaseAnchorForCommentsOnly` is used only for rendering messages. Feel free to let me know if I missed anything or if you want me to test any particular use case/issue more.

**Comment by mollfpr** [comment]
@Nikhil-Vats Sorry I didn't get the time to test this, but I'll try to complete the review in the morning.

**Comment by Nikhil-Vats** [comment]
> @Nikhil-Vats Is this expected in iOS to open the link when clicking on the space?
> 
> Also, the emoji got cut out in this PR but not in the latest main.

@mollfpr The links on iOS do open when we click on the space. This happens on the prod app and the main branch as well. The link padding/click issues that this PR is supposed to fix are only for the web environment so I think it is expected behaviour on iOS.

Regarding, the emoji issue. It exists on the main branch and the prod app as well but the emoji should be part of the heading/code (`code with 😀`). In your screenshot for the main branch, the emoji is not cut because they are not part of the heading/code.

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



19545 Web

https://github.com/Expensify/App/assets/25520267/cff3bf45-f031-4201-a153-12f0f1d916eb


</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://github.com/Expensify/App/assets/25520267/b55c6c09-6ab6-42e9-ae16-2aec26a3a48f


https://github.com/Expensify/App/assets/25520267/f1e740b2-5640-4c65-9d88-87237c70f8a7


</details>

<details>
<summary>Mobile Web - Safari</summary>



https://github.com/Expensify/App/assets/25520267/86ad33b5-b7eb-41ad-90cb-94dfe22276e9


https://github.com/Expensify/App/assets/25520267/fd259136-fd8d-40c9-9ecd-c107f97390d1


</details>

<details>
<summary>Desktop</summary>



19545 Desktop


https://github.com/Expensify/App/assets/25520267/35498384-e083-45d9-b033-4309be58099d


</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



19545 Android


https://github.com/Expensify/App/assets/25520267/9c192871-8349-46da-a0cd-b4933f779691


</details>

**Comment by mollfpr** [comment]
> For attachments without previews like CSV, docs, etc. A strikethrough should be shown similar to text comments.

@Nikhil-Vats The attachment text strikethrough is only for web, no?

Simulator Screenshot - iPhone 14 - 2023-05-30 at 22 35 05

**Comment by Nikhil-Vats** [comment]
> @Nikhil-Vats The attachment text strikethrough is only for web, no?

@mollfpr yes, the behaviour on phone for all attachments is just reduced opacity.

**Review by mollfpr — DISMISSED** [review]
@Nikhil-Vats Really great test flow presentation 👍

All the objection I found is already address and seems the test step it's covered enough. I'll go a head approve this PR.

All yours @cristipaval @parasharrajat

File: src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
Comment (Nikhil-Vats) [reply src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js:51]: Hey @parasharrajat, can you please tell me what you mean by this? Do you mean moving the `inline` prop or the `style` prop?

**Review by mollfpr — APPROVED** [review]

File: src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
Comment (Nikhil-Vats) [reply src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js]: Hey @parasharrajat, friendly bump! 😄

**Review by cristipaval — DISMISSED** [review]

**Comment by Nikhil-Vats** [comment]
Hey @mollfpr, do we need any more approvals or are we good to go?

**Comment by mollfpr** [comment]
I tend to wait for @parasharrajat final review.

**Comment by mollfpr** [comment]
Bump @parasharrajat

**Comment by mollfpr** [comment]
@cristipaval I think we can merge this. I see that all the concerns have already been addressed.

**Comment by Nikhil-Vats** [comment]
Updated the PR after merge conflicts. New test proofs -

<details>
<summary>
Web
</summary>

For this issue, link padding issue, multiline comment issue - 
https://github.com/Expensify/App/assets/31413407/e900ad92-fede-4a56-8a75-dc3756ef4407

For [attachment issue](https://github.com/Expensify/App/issues/19717) (attachments are downloadable when we click on right of the button) -
https://github.com/Expensify/App/assets/31413407/2a708870-fc44-4096-a22e-95ebf6a863c7


</details>

**Comment by Nikhil-Vats** [comment]
@mollfpr @cristipaval There were merge conflicts since PR for https://github.com/Expensify/App/issues/19717 was merged in main. This issue was also a regression from the [same PR](https://github.com/Expensify/App/pull/17496/files) that caused this one so they were closely related. 

So, I talked with @bernhardoj who worked on the other issue and also added test proofs for their issue (attachments and images were clickable in empty space on right) in my latest comment.

So my latest comment includes test proofs for -
1. this issue - https://github.com/Expensify/App/issues/18658
2. attachments clickable issue - https://github.com/Expensify/App/issues/1971
3. the original issue that created 1 and 2 - https://github.com/Expensify/App/issues/16526
4. links in header have extra padding which is clickable - https://github.com/Expensify/App/issues/17488

File: src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js
Comment (Nikhil-Vats) [inline src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js:63]: Also, there was a typo here since the name of prop for `PressableWithoutFocus` is `styles` not `style`. 

https://github.com/Expensify/App/blob/213ba5509a51a812bed795d182d65da01f1d13e8/src/components/PressableWithoutFocus.js#L17

I noticed this while reviewing the PR for #19717 so I am fixing it here.

**Review by cristipaval — APPROVED** [review]
Thanks @Nikhil-Vats for keeping us updated and for the rigorous testing!

File: src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js
Comment (bernhardoj) [reply src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js:63]: I get this console error:
image

Comment (Nikhil-Vats) [reply src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js:63]: @parasharrajat can you link the PR if possible? 
Because if that’s a big one and will take time to be merged, I can raise a PR and fix this sooner.

Comment (Nikhil-Vats) [reply src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js:63]: @mollfpr @cristipaval 
`styles` has to be an array here and I passed an object.

I made this typo while fixing an issue created in a related PR. Shall I create a new PR for this so that it is fixed before QA testing? Otherwise if the other PR that @parasharrajat is talking about takes a lot of time then this PR will be blocked unnecessarily.

File: src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js
Comment (Nikhil-Vats) [reply src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js:57]: @mollfpr @cristipaval This PR was reverted because of the special handling of internal links explained by @parasharrajat above.

I found a solution -
basically for internal links since they are rendered as `div` we can explicitly set `cursor` to `pointer` on line 71. For normal links it is added by default so the only behaviour we are changing is for internal links by adding pointer cursor -
(Text below uses `div` for internal links and `a` for other/normal links)
https://github.com/Expensify/App/blob/6e8435ec6cba470dae50262439f2bacecea9222f/src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js#L68-L71

Need to change line 71 to -
```javascript
style={StyleSheet.flatten([props.style, defaultTextStyle, styles.cursorPointer])}
```


This would solve all the related issues without regression. I tested on local all issues that were related to my issue like https://github.com/Expensify/App/issues/18658, https://github.com/Expensify/App/issues/17488, #16526, are fixed.

Result -

https://github.com/Expensify/App/assets/31413407/808e6e2f-371a-4e68-be19-46b02cf469d0

**Pull request merged**
